scripts/data_files/query_config.fmt

```diff
@@ -99,9 +99,9 @@
 #define MACRO_NAME_TO_STR(macro)                                        \
     mbedtls_printf( "%s", strlen( #macro "" ) > 0 ? #macro "\n" : "" )
 
-#define NAME_TO_STR(macro)  #macro
+#define STRINGIFY(macro)  #macro
 #define OUTPUT_MACRO_NAME_VALUE(macro) mbedtls_printf( #macro "%s\n",   \
-    strlen( NAME_TO_STR(macro) "") > 0 ? "=" NAME_TO_STR(macro) : "" )
+    STRINGIFY(macro)[0] != 0 ? "=" STRINGIFY(macro) : "" )
```

Comment:
MSVC says:
```
    52>..\..\programs\test\query_config.c(2500): error C2059: syntax error : '[' [C:\builds\workspace\mbed-tls-pr-head_PR-5303-head\worktrees\tmpposg_ka7\visualc\VS2010\ssl_client2.vcxproj]
```
(and again each time this macro is used). I don't see the problem, but MSVC's preprocessor is not completely standards-compliant, so maybe we've hit a corner case? Maybe `(STRINGIFY(macro))[0]` would work?

Reply:
fixed with `(STRINGIFY(macro) "" )[0]`